lunr/match_data.py

```diff
@@ -64,5 +67,7 @@ def add(self, term, field, metadata):
             else:
                 self.metadata[term][field][key] = metadata[key]
 
-    def __eq__(self, other):
+    def __eq__(self, other: object):
```

Comment:
Why is this not also `MatchData` as above?

Reply:
I feel like this should be documented in `mypy`'s documentation somewhere but I can't figure out where!  It's also explained here: https://stackoverflow.com/questions/54801832/mypy-eq-incompatible-with-supertype-object